### ISSUE TITLE
P1 audit v4: condition-mediated mortality RNG assignment is sensitive to arbitrary founder person labels

### ISSUE THREAD

**Issue by Datengu** [Datengu/anthrosim#497]
## Scientific Audit v4 finding — AV4-006

**Severity:** P1  
**Primary Area:** A — authoritative semantics and scheduler behaviour  
**Cross-cutting:** B — demography; D — resources/condition; E — spatial attribution; H — stochasticity/RNG; N — cross-system integration

## Affected immutable baseline

- release: `v0.3.4`
- exact tag SHA: `8996e99ffc4c5b91b9e00d1048eedd4227ea1d09`
- model semantics: `anthrosim-model-semantics-v25`

## Fresh Audit-v4 executable reproduction

Evidence PR #496, exact evidence head `21775e28406d9c6af803aeaa8cbbb595a1eb3e2d`, dedicated workflow run `33690381210`, job `100447530200`.

Test: `crates/anthrosim-core/tests/audit_v4_condition_mortality_relabel.rs`

Construction:
- two fixed one-person households at `CellId(1)` and `CellId(2)`;
- otherwise-equivalent 30-year-old male founders, each at condition 0;
- arm B exchanges only which canonical `PersonId` occupies which fixed household/cell;
- after erasing canonical labels, the scientific state is identical;
- same seed;
- background demographic mortality = 0;
- fertility = 0;
- one M3 resource/condition period per year;
- annual resource need = 0;
- condition recovery and condition loss = 0, so condition remains fixed at 0;
- max condition-mediated mortality probability = `150,000 / 1,000,000`;
- migration disabled;
- one-year horizon.

The test is specified across seeds 1..1000 but fails immediately at seed 1.

Exact observed result:

```text
A=[CellId(2)]
B=[CellId(1)]
```

Pinned Rust 1.97.1 compiled and ran the test successfully; the red result is the intended scientific equality assertion, not setup/compilation failure.

## Mechanism

M3 condition-mediated mortality uses its own deterministic RNG stream but consumes draws while iterating canonical population records at the mortality boundary. A pure relabelling therefore changes which fixed spatial founder receives a given same-seed condition-mortality realization. This is distinct from AV4-002/#488, which concerns the separate M2/background demographic-mortality stream.

## Scientific consequence and severity

This is P1 because arbitrary bookkeeping identity changes which spatially situated person dies while all scientific state and the seed are held fixed. Condition-mediated death directly affects household composition, future resource demand, fertility/parent availability, kin topology, migration, temporary mobility and spatial observables. Same-seed paired or counterfactual comparisons can therefore diverge because of canonical record order rather than the scientific intervention.

## Discovery-phase rule

Do **not** repair during Audit-v4 discovery. Preserve this finding and continue Areas A–N against immutable v0.3.4/v25.

## Required post-discovery repair contract

After discovery completes, condition-mediated mortality stochastic realization must not depend on arbitrary canonical record ordering for scientifically equivalent states. Acceptance must include:
- the exact relabel adversary passing over a substantial multi-seed range;
- no replacement with another arbitrary storage/container ordering;
- condition-to-risk semantics preserved;
- competing-risk resolution/cause attribution remains correct when background mortality is also active;
- explicit stream separation and deterministic provenance preserved;
- checkpoint/replay equivalence remains exact;
- downstream household/spatial state remains invariant under scientifically equivalent relabelling;
- protected CI and applicable scientific/security gates pass;
- independent post-merge adversarial re-verification because this is P1.

**Comment by Datengu** [comment]
AV4-006 post-discovery remediation is now independently verified on merged `main`.

## Production repair
- production PR: #568
- exact final production head: `348cd13dc0b1efe60fe8414ae0d7ed03dd9c8d13`
- exact production merge SHA: `bce11b93de8f227bec987f62df8357515ad5707b`
- model semantics: `anthrosim-model-semantics-v30`
- checkpoint schema: `18`
- exact central CI run: `33824195793` — success
- Quality and tests job: `100873588434` — success
- Release build job: `100875277389` — success
- Core benchmarks job: `100875277412` — success
- protected Canonical M7.6 job: `100875993151` — success
- final-head M7.6 artifact: `9919676973`, SHA-256 `acdd78a3ca425fc01152845f07f4c7aaf0737e44ccb5541cc3037b8e661ece1d`

The production repair removes arbitrary canonical `PersonId` / packed-record order from condition-mediated mortality realization assignment by scheduling mortality candidates with persistent person `stochastic_coupling_rank`. The same schedule is also used for background latent triggers and simultaneous-trigger cause attribution, while preserving the existing condition-to-risk mathematics, independent named condition/background RNG streams, exchange-symmetric competing-risk resolver, mortality boundary semantics, and deterministic stream-position/checkpoint accounting.

Permanent regression `crates/anthrosim-core/tests/condition_mortality_label_invariance.rs` covers the original-style 1..=1000 pure founder-PersonId relabel sweep, three-person cyclic relabels over 256 seeds, certain simultaneous condition/background risk over 256 seeds, and two-year downstream death/living-cell/resource propagation over 256 seeds.

## Deliberate downstream scientific re-verification
- #304 demographic baseline review: run `33820753145`, job `100862690900`, artifact `9918271667`, SHA-256 `76a943bae2a8a3bef13ee551f17edeb6d485897b03a52d8607ebe3deeb3bb634`; 780/780 runs completed, all three Monte Carlo precision gates returned `sufficient_stop`, and the recommendation remained `no_universal_demographic_baseline`. The exact final production head independently reconfirmed this in run `33824195815`, job `100873227835`, artifact `9919446721`, SHA-256 `66d29476b7c9d69a925e6b0e6480c2b742976178070604f77fbfe4d8b80041f7`.
- M7.6 review: run `33821069979`, job `100863637973`, artifact `9918382546`, SHA-256 `3f27490d2c30098aab71ccc83d4f0ad2f822af708503694dae5b7f525cdd3920`; 144/144 canonical runs completed. Exact point values changed under v30, but all predeclared synthetic-validation contrasts were preserved: all three productivity-250 no-migration arms remained extinct 8/8, matched migration-enabled arms persisted 8/8, and migration-enabled runs retained higher terminal population, lower condition mortality and lower unmet resource need at all 9/9 matched points. The deliberately rebaselined v30 reference then passed the exact final production head in central CI job `100875993151`; its semantics guard now reads the authoritative `MODEL_SEMANTICS_ID`, while the fail-closed 18-point equality assertion remains unchanged.
- M8.6 review: run `33820753353`, job `100862718819`, artifact `9918254916`, SHA-256 `da9a934eb46176d7f8acc4e74f08a628ac72881c0756e7840a365420ea4ed292`; classification remained `fragile_spatial_structure`, with no robust metrics and the same two fragile metrics. Final-head applicable-gate run `33824195966`, job `100873303898` passed source validation, four ensembles, observability, aggregation, canonical v30 reference, tamper rejection, and archive upload. Final-head artifact `9919427470`, SHA-256 `96de47ed541cf9c08ce3a79af47f512f44bc8a26b19f4b1043efdfd4efe9f8b6`.
- M9.7 required no machine-reference change. Final-head applicable-gate run `33824195966`, job `100873303891` passed paired ensembles, observability, exact replay, active checkpoint/resume, preserved reference, tamper rejection, and archive upload. Final-head artifact `9919448520`, SHA-256 `5b1b57e8b66f0e41c45f0b205dd9eb957cfa80f2134b95ef86bf3d75f35fac10`.

No threshold, seed set, experiment design, acceptance criterion, or scientific gate was weakened.

## Independent post-merge original adversary
GitHub auto-closed this issue on production merge, so it was reopened before evidence execution.

Evidence PR #569 was created from exact merged `main` `bce11b93de8f227bec987f62df8357515ad5707b` and is now closed **unmerged**.

The exact immutable #496 adversary blobs were restored byte-for-byte:
- test blob `5c72fb7dfafcd09eee7f58394285474f22b718d9`
- workflow blob `8cf9e48b107d846d4d6e9932cf176b2d38bc8870`
- evidence head `9fd8bcc2aa6d85bc0a7395bde3a3176bb05c8e73`
- workflow run `33825847385`
- job `100878214240`
- Rust: pinned `1.97.1`
- exact original command: `cargo test -p anthrosim-core --test audit_v4_condition_mortality_relabel --locked -- --nocapture`
- exact job result: `1 passed; 0 failed`
- original internal seed sweep: `1..=1000`
- post-merge scientific divergence: **0/1000**

For comparison, immutable discovery PR #496 / head `21775e28406d9c6af803aeaa8cbbb595a1eb3e2d`, run `33690381210`, job `100447530200` failed immediately at seed 1 with A=`[CellId(2)]`, B=`[CellId(1)]`.

Acceptance contract is satisfied. No collateral closure is claimed for AV4-007/#500 or any later Audit-v4 finding.
